examples/sql_bank/main.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/lib/pq"
+
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const (
+	maxTransfer = 999
+	numAccounts = 3
+	concurrency = 2
+	aggregate   = true
+)
+
+var (
+	dbName      = "bank"
+	tableName   = pq.QuoteIdentifier("accounts")
+	idName      = pq.QuoteIdentifier("id")
+	balanceName = pq.QuoteIdentifier("balance")
+)
+
+func moveMoney(db *sql.DB) {
+	for {
+		from, to := rand.Intn(numAccounts), rand.Intn(numAccounts)
+		if from == to {
+			continue
+		}
+		amount := rand.Intn(maxTransfer)
+		tx, err := db.Begin()
+		if err != nil {
+			log.Fatal(err)
+		}
+		var fromBalance, toBalance int
+		if err = tx.QueryRow(fmt.Sprintf("SELECT %s FROM %s WHERE id=$1", balanceName, tableName), from).Scan(&fromBalance); err != nil {
+			log.Fatal(err)
+		}
+		if err = tx.QueryRow(fmt.Sprintf("SELECT %s FROM %s WHERE id=$1", balanceName, tableName), to).Scan(&toBalance); err != nil {
+			log.Fatal(err)
+		}
+
+		// TODO: remove
+		log.Infof("amt: %3d; src: %1d, dst: %1d, srcBal: %5d, dstBal: %5d", amount, from, to, fromBalance, toBalance)
+
+		if _, err = tx.Exec(fmt.Sprintf("UPDATE %s SET %s=$2 WHERE id=$1", tableName, balanceName), from, fromBalance-amount); err != nil {
+			log.Warning(err)
+			if err = tx.Rollback(); err != nil {
+				log.Fatal(err)
+			}
+			continue
+		}
+		if _, err = tx.Exec(fmt.Sprintf("UPDATE %s SET %s=$2 WHERE id=$1", tableName, balanceName), to, toBalance+amount); err != nil {
```

Comment:
It's unfortunate that most database drivers don't support batching, because it would be much nicer to send the two updates in a batch than to contort the SQL into a single statement. 

Also, the goal is not necessarily to maximize throughput; we want to see what happens with multi-step operations (for example, the KV bank example uses Put instead of Increment).